src/components/StarknetDapp.tsx

```diff
@@ -13,10 +13,21 @@ import { SectionButton } from "./sections/SectionButton"
 import { SectionLayout } from "./sections/SectionLayout"
 import { SessionKeysSign } from "./sections/SessionKeys/SessionKeysSign"
 import { Section } from "./sections/types"
+import { DeclareContract } from "./sections/Declare/DeclareContract"
+import { useSearchParams } from "next/navigation"
+import { upperFirst } from "@/helpers/upperFirst"
 
-const StarknetDapp = () => {
-  const [section, setSection] = useState<Section | undefined>(undefined)
+const StarknetDappContent = () => {
+  const [section, setSection] = useState<Section | undefined>("Connection")
```

Comment:
@dhruvkelawala these changes are causing the tests to fail

I would keep it as it was before
- state starting as `undefined`

and removing this block

```typescript
const searchParams = useSearchParams()

  useEffect(() => {
    const section = searchParams.get("section")
    if (section && isConnected) {
      setSection(upperFirst(section) as Section)
    }
  }, [searchParams, isConnected])
```


it's also the behaviour we want to keep with the demo dapp (no keeping the section opened when refreshing)

Reply:
I wanted to redirect user to Declare section when user click on `Declare & Deploy` on AX